oceanprefilter/oceanprefilter.go

```
@@ -305,33 +303,16 @@ func (pf *prefilter) ClassificationsFromCamera(
 func (pf *prefilter) Classifications(ctx context.Context, img image.Image,
 	n int, extra map[string]interface{},
 ) (classification.Classifications, error) {
-	if pf.camName == "" {
-		isTriggered, err := make_inference(img, pf.rc)
-		if err != nil {
-			pf.logger.Infow("classification error", "error", err.Error())
-		}
-		cls := []classification.Classification{}
-		if isTriggered {
-			c := classification.NewClassification(1.0, triggerClassName)
-			cls = append(cls, c)
-		}
-		return classification.Classifications(cls), nil
-
-	} else {
-		select {
-		case <-ctx.Done():
-			return nil, errors.Wrap(ctx.Err(), "module might be configuring")
-		case <-pf.cancelContext.Done():
-			return nil, errors.Wrap(pf.cancelContext.Err(), "lost connection with background camera stream loop")
-		default:
-			cls := []classification.Classification{}
-			if pf.triggerFlag.Load() {
-				c := classification.NewClassification(1.0, triggerClassName)
-				cls = append(cls, c)
-			}
-			return classification.Classifications(cls), nil
-		}
+	isTriggered, err := MakeInference(img, pf.rc)
+	if err != nil {
+		pf.logger.Infow("classification error", "error", err.Error())
```

You can return an error as normal rather than just using the logger here